pkg/kallsyms/kallsyms.go

```diff
@@ -181,17 +187,25 @@ func specUpdateAddresses(
 
 	// Add the symbols that are not in the cache to the cache
 	if !allFoundInCache {
-		k, err := kAllSymsFactory()
-		if err != nil {
-			triedGetAddr[symbols[0]] = err
-			return err
+		// initialize the resolvers
+		for _, resolver := range symbolResolvers {
+			if err := resolver.init(); err != nil {
+				return fmt.Errorf("initializing resolver: %w", err)
```

Comment:
This will initialize both the kallsyms resolver and the ebpf resolver. But it might be that the first one works fine so there is no need to initialize the second one.
In practice, the ebpf resolver initialization is a no-op and it is at the second place, so it does not impact performance, but it makes the code looks weird.

Also, if the kernel has `CONFIG_KALLSYMS=n`, `/proc/kallsyms` will not exist and kallsyms resolver's init() will return an error and it will not fallback to the ebpf resolver.

So maybe init() should not return an error:
```diff
 type symbolResolver interface {
-	init() error
+	init()
```
The error could be saved in the `kAllSymsStrResolver struct` and returned later at `resolve()`-time:
```diff
 type kAllSymsStrResolver struct {
 	kAllSymsResolver
 	KAllSymsStr string
+	initErr error
 }
```
What do you think?